network/message.go

```diff
@@ -0,0 +1,142 @@
+package network
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/imdario/mergo"
+	"github.com/nknorg/nconnect/admin"
+	"github.com/nknorg/nconnect/config"
+	"github.com/nknorg/nkn-sdk-go"
+	"github.com/nknorg/nkngomobile"
+)
+
+// msgType constants
+const (
+	MT_NONE                = 0
+	JOIN_NETWORK           = 1
+	UPDATE_MY_INFO         = 2
+	GET_MY_INFO            = 3
+	GET_NODES_I_ACCEPT     = 4
+	GET_NODES_I_CAN_ACCESS = 5
+	LEAVE_NETWORK          = 6
+
+	NOTI_AUTHORIZED    = 7
+	NOTI_NEW_MEMBER    = 8
+	NOTI_SET_ACCEPT    = 9
+	NOTI_MEMBER_ONLINE = 10
+)
+
+type nodeInfo struct {
+	IP       string    `json:"ip"`
+	Netmask  string    `json:"netmask"`
+	Name     string    `json:"name"`
+	Address  string    `json:"address"`
+	LastSeen time.Time `json:"lastSeen"`
+}
+
+type networkInfo struct {
+	Domain  string `json:"domain"`
+	Gateway string `json:"gateway"`
+	DNS     string `json:"dns"`
+}
+
+type memberToManager struct {
+	MsgType int    `json:"msgType"`
+	Name    string `json:"name"`
+}
+
+type managerToMember struct {
+	MsgType     int          `json:"msgType"`
+	Err         error        `json:"err"`
+	NetworkInfo *networkInfo `json:"networkInfo"`
+	NodeInfo    []*nodeInfo  `json:"nodeInfo"`
+}
+
+func SendMsg(mc *admin.Client, address string, msg interface{}, waitResponse bool) (*managerToMember, error) {
+	reqBytes, err := json.Marshal(msg)
+	if err != nil {
+		return nil, err
+	}
+
+	var onMessage *nkn.OnMessage
+	for i := 0; i < 3; i++ {
+		onMessage, err = mc.Send(nkn.NewStringArray(address), reqBytes, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		if !waitResponse {
+			return nil, nil
+		}
+
+		select {
+		case resp := <-onMessage.C:
+			var respMsg managerToMember
+			if err = json.Unmarshal(resp.Data, &respMsg); err != nil {
+				return nil, err
+			}
+			return &respMsg, nil
+
+		case <-time.After(10 * time.Second):
+			err = admin.ErrReplyTimeout
+		}
+	}
+
+	return nil, err
+}
+
+func GetMultiClient(opts *config.Opts) (*nkn.MultiClient, error) {

```

Comment:
Feels like there are lots of redundant code here

Reply:
Yes. removed.